File: frontend/src/types/asset.ts
export type AssetStatus = 'IN_USE' | 'RENTED' | 'BROKEN' | 'IN_STORAGE' | 'DISPOSED';

export interface Asset {
  assetId: number;
  assetCode: string;
  assetName: string;
  status: AssetStatus;
  categoryName: string;
  categoryId: number;
  location: string | null;
  managingDepartment: string | null;
  usingDepartment: string | null;
  manufacturer: string | null;
  modelNumber: string | null;
  purchaseDate: string | null;
  imagePath: string | null;
  aiClassified: boolean;
  createdAt: string;
}

export interface AssetDetail {
  assetId: number;
  assetCode: string;
  assetName: string;
  status: AssetStatus;
  categoryId: number;
  categoryName: string;
  categoryPath: string[];
  serialNumber: string | null;
  manufacturer: string | null;
  modelNumber: string | null;
  purchaseDate: string | null;
  location: string | null;
  managingDepartment: string | null;
  usingDepartment: string | null;
  conditionRating: number | null;
  technicalSpecs: string | null;
  imagePath: string | null;
  aiClassified: boolean;
  notes: string | null;
  registeredByName: string;
  createdAt: string;
  updatedAt: string;
}

export interface AssetCreateRequest {
  categoryId: number;
  assetName: string;
  serialNumber?: string;
  manufacturer?: string;
  modelNumber?: string;
  purchaseDate?: string;
  location?: string;
  managingDepartment?: string;
  usingDepartment?: string;
  conditionRating?: number;
  technicalSpecs?: string;
  aiClassified?: boolean;
  notes?: string;
}

export interface AssetUpdateRequest {
  categoryId: number;
  assetName: string;
  serialNumber?: string;
  manufacturer?: string;
  modelNumber?: string;
  purchaseDate?: string;
  location?: string;
  managingDepartment?: string;
  usingDepartment?: string;
  conditionRating?: number;
  technicalSpecs?: string;
  notes?: string;
}

export interface AssetStatusRequest {
  status: AssetStatus;
}

export interface AssetSummary {
  total: number;
  inUse: number;
  rented: number;
  broken: number;
  inStorage: number;
  disposed: number;
}

export interface AssetSearchParams {
  page?: number;
  size?: number;
  status?: AssetStatus;
  categoryId?: number;
  search?: string;
  location?: string;
  managingDepartment?: string;
  usingDepartment?: string;
  sort?: string;
}

File: mobile/lib/features/asset/data/models/asset.dart
import 'asset_status.dart';

class Asset {
  final int assetId;
  final String assetCode;
  final String assetName;
  final AssetStatus status;
  final String categoryName;
  final int categoryId;
  final String? location;
  final String? managingDepartment;
  final String? usingDepartment;
  final String? manufacturer;
  final String? modelNumber;
  final DateTime? purchaseDate;
  final String? imagePath;
  final bool aiClassified;
  final DateTime createdAt;

  const Asset({
    required this.assetId,
    required this.assetCode,
    required this.assetName,
    required this.status,
    required this.categoryName,
    required this.categoryId,
    this.location,
    this.managingDepartment,
    this.usingDepartment,
    this.manufacturer,
    this.modelNumber,
    this.purchaseDate,
    this.imagePath,
    required this.aiClassified,
    required this.createdAt,
  });

  factory Asset.fromJson(Map<String, dynamic> json) {
    return Asset(
      assetId: json['assetId'] as int,
      assetCode: json['assetCode'] as String,
      assetName: json['assetName'] as String,
      status: AssetStatus.fromValue(json['status'] as String),
      categoryName: json['categoryName'] as String,
      categoryId: json['categoryId'] as int,
      location: json['location'] as String?,
      managingDepartment: json['managingDepartment'] as String?,
      usingDepartment: json['usingDepartment'] as String?,
      manufacturer: json['manufacturer'] as String?,
      modelNumber: json['modelNumber'] as String?,
      purchaseDate: json['purchaseDate'] != null
          ? DateTime.parse(json['purchaseDate'] as String)
          : null,
      imagePath: json['imagePath'] as String?,
      aiClassified: json['aiClassified'] as bool? ?? false,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
